csharp/src/Ice/OutputStream.cs

```diff
@@ -264,6 +324,90 @@ public void WriteString(string v)
             }
         }
 
+        /// <summary>Writes a uint to the stream.</summary>
+        /// <param name="v">The uint to write to the stream.</param>
+        public void WriteUInt(uint v) => WriteFixedSizeNumeric(v);
+
+        /// <summary>Writes a ulong to the stream.</summary>
+        /// <param name="v">The long to write to the stream.</param>
+        public void WriteULong(ulong v) => WriteFixedSizeNumeric(v);
+
+        /// <summary>Writes a ushort to the stream.</summary>
+        /// <param name="v">The ushort to write to the stream.</param>
+        public void WriteUShort(ushort v) => WriteFixedSizeNumeric(v);
+
+        /// <summary>Writes an int to stream, using Ice's variable-length encoding.</summary>
+        /// <param name="v">The int to write to the stream.</param>
+        public void WriteVarInt(int v) => WriteVarLong(v);
+
+        /// <summary>Writes a long to stream, using Ice's variable-length encoding, with the minimum number of bytes
+        /// required by the encoding.</summary>
+        /// <param name="v">The long to write to the stream. It must be in the range [2^61..2^61 - 1].</param>
+        public void WriteVarLong(long v)
+        {
+            if (v < EncodingDefinitions.VarLongMinValue || v > EncodingDefinitions.VarLongMaxValue)
+            {
+                throw new ArgumentOutOfRangeException($"varlong value `{v}' is out of range", nameof(v));
+            }
+
+            if (OldEncoding)
+            {
+                WriteLong(v);
+            }
+            else
+            {
+                v <<= 2;
+                int encodedLength = v switch
+                {
+                    long b when b >= sbyte.MinValue && b <= sbyte.MaxValue => 0,
+                    long s when s >= short.MinValue && s <= short.MaxValue => 1,
+                    long i when i >= int.MinValue && i <= int.MaxValue => 2,
+                    _ => 3
+                };
+
+                v |= (uint)encodedLength;
+                Span<byte> data = stackalloc byte[sizeof(long)];
+                MemoryMarshal.Write(data, ref v);
+                WriteByteSpan(data.Slice(0, 1 << encodedLength));
+            }
+        }
+
+        /// <summary>Writes a uint to stream, using Ice's variable-length encoding.</summary>
```

Comment:
"variable-length encoding" or "variable-length integer encoding", InputStream uses the later.

Reply:
Fixed.